src/Util/YamlSourceManipulator.php

```diff
@@ -771,16 +792,26 @@ private function findEndPositionOfValue($value, $offset = null)
         }
 
         if (is_scalar($value) || null === $value) {
+            $offset = null === $offset ? $this->currentPosition : $offset;
+
             if (\is_bool($value)) {
                 // (?i) & (?-i) opens/closes case insensitive match
                 $pattern = sprintf('(?i)%s(?-i)', $value ? 'true' : 'false');
             } elseif (null === $value) {
                 $pattern = '(~|NULL|null|\n)';
             } else {
-                $pattern = sprintf('\'?"?%s\'?"?', preg_quote($value, '#'));
-            }
+                // Multiline value ends with \n.
+                // If we remove this character, the next property will ne merged with this value
+                $quotedValue = preg_quote(rtrim($value, "\n"), '#');
+                $patternValue = $quotedValue;
+
+                // Iterates until we find a new line char or we reach end of file
+                if ($this->findPositionOfMultilineCharInLine($offset)) {
```

Comment:
```suggestion
                if ($this->findPositionOfMultilineCharInLine($offset) !== null) {
```

That will help clarify that we're looking for IF there is a multiline char here, we're not (in this case) actually caring about the exact position returned.